### PR TITLE
NamedFile using Body::Sized with metadata len

### PR DESCRIPTION
**Pull request by wpbirney**
NamedFile using the files metadata length instead of seeking

### CODE REVIEW

File: lib/src/response/named_file.rs [hunk 1]
```diff
@@ -87,8 +88,9 @@ impl Responder<'static> for NamedFile {
                 response.set_header(ct);
             }
         }
-
-        response.set_streamed_body(BufReader::new(self.take_file()));
+        let len = self.0.metadata()
```
Comment (SergioBenitez) [inline lib/src/response/named_file.rs]: Don't `panic!` here. Instead, fall-back to a streamed file if this fails.

**Comment by zxvfxwing** [comment]
Thank you for this pull request (i'm the spoken from matrix/IRC channel).
I'm glad I helped you to pointed this issue.

I will now have the remaining time displayed in my navigator when downloading files from Rocket 👍

File: lib/src/response/named_file.rs [hunk 2]
```diff
@@ -87,8 +88,10 @@ impl Responder<'static> for NamedFile {
                 response.set_header(ct);
             }
         }
-
-        response.set_streamed_body(BufReader::new(self.take_file()));
+        match self.0.metadata() {
```
Comment (SergioBenitez) [inline lib/src/response/named_file.rs]: Please _add_ an empty line above here.

File: lib/src/response/named_file.rs [hunk 3]
```diff
@@ -87,8 +88,10 @@ impl Responder<'static> for NamedFile {
...
+            Ok(meta) => response.set_raw_body(Body::Sized(self.take_file(), meta.len())),
+            Err(_) => response.set_streamed_body(self.take_file())
+        }
```
Comment (SergioBenitez) [inline lib/src/response/named_file.rs]: Please _add_ an empty line below here.

Comment (wpbirney) [reply lib/src/response/named_file.rs]: formatting changes pushed

**Comment by SergioBenitez** [comment]
The build is breaking due to https://github.com/jgallagher/rusqlite/issues/313.

**Comment by SergioBenitez** [comment]
Merged in 4749b541a9ba1e09b212ce926bb59171092982e9. Thanks so much! :)

**Pull request closed without merging**